src/Cms/PluginAssets.php

```diff
@@ -42,6 +49,73 @@ public static function clean(string $pluginName): void
 		}
 	}
 
+	/**
+	 * Filters assets collection by CSS files
+	 */
+	public function css(): static
+	{
+		return $this->filter(fn ($asset) => $asset->extension() === 'css');
+	}
+
+	/**
+	 * Creates a new collection for the plugin's assets
+	 * by considering the plugin's `asset` extension
+	 * (and `assets` directory as fallback)
+	 */
+	public static function factory(Plugin $plugin): static
+	{
+		// get assets defined in the plugin extension
+		if ($assets = $plugin->extends()['assets'] ?? null) {
+			if ($assets instanceof Closure) {
+				$assets = $assets();
+			}
+
+			// normalize array: use relative path as
+			// key when no key is defined
+			foreach ($assets as $key => $root) {
+				if (is_int($key) === true) {
+					unset($assets[$key]);
+					$path = Str::after($root, $plugin->root() . '/');
+					$assets[$path] = $root;
+				}
+			}
+		}
+
+		// fallback: if no assets are defined in the plugin extension,
+		// use all files in the plugin's `assets` directory
+		if ($assets === null) {
+			$assets = [];
+			$root   = $plugin->root() . '/assets';
+
+			foreach (Dir::index($root, true) as $path) {
+				if (is_file($root . '/' . $path) === true) {
+					$assets[$path] = $root . '/' . $path;
+				}
+			}
+		}
+
+		$collection = new static([], $plugin);
+
+		foreach ($assets as $path => $root) {
+			$collection->$path = new PluginAsset($path, $root, $plugin);
```

Comment:
To avoid the magic setter:

```suggestion
			$collection->data[$path] = new PluginAsset($path, $root, $plugin);
```